PocketKit/Sources/PocketKit/Root/RootCoordinator.swift

```diff
@@ -41,6 +42,64 @@ class RootCoordinator {
         }.store(in: &subscriptions)
 
         window?.makeKeyAndVisible()
+        
+        rootViewModel.$bannerViewModel.sink { [weak self] viewModel in
+            if let viewModel = viewModel {
+                self?.setupBanner(with: viewModel)
+            } else {
+                self?.dismissView()
+            }
+        }.store(in: &subscriptions)
+
+        NotificationCenter.default.publisher(for: UIApplication.didEnterBackgroundNotification).sink { [weak self] _ in
+            self?.removeBanner()
+        }.store(in: &subscriptions)
+    }
+    
+    private func setupBanner(with viewModel: BannerViewModel) {
+        let bannerView = BannerView()
+        bannerView.configure(model: viewModel)
+
+        window?.addSubview(bannerView)
+        configureConstraints(bannerView)
+        
+        let tabBarHeight: CGFloat = main?.tabBar?.frame.height ?? 0
+        let translation = window?.traitCollection.userInterfaceIdiom == .phone ? tabBarHeight : 0
+      
+        UIView.animate(withDuration: 0.5, delay: 0, options: [.allowUserInteraction], animations: {
+            bannerView.transform = CGAffineTransform(translationX: 0, y: -translation - 12)
+        })
+
+        self.bannerView = bannerView
+    }
+    
+    private func removeBanner() {
+        bannerView?.removeFromSuperview()
+        bannerView = nil
+    }
+    
+    private func configureConstraints(_ bannerView: UIView) {
+        guard let window = window else { return }
+        bannerView.translatesAutoresizingMaskIntoConstraints = false
+        if window.traitCollection.userInterfaceIdiom == .pad {
+            bannerView.bottomAnchor.constraint(equalTo: window.bottomAnchor, constant: -12).isActive = true
+            bannerView.widthAnchor.constraint(equalToConstant: 600).isActive = true
+            bannerView.centerXAnchor.constraint(equalTo: window.centerXAnchor).isActive = true
+        } else {
+            let tabBar = main?.tabBar ?? window
+            bannerView.bottomAnchor.constraint(equalTo: tabBar.bottomAnchor, constant: 0).isActive = true
+            bannerView.leadingAnchor.constraint(equalTo: window.leadingAnchor, constant: 12).isActive = true
+            bannerView.trailingAnchor.constraint(equalTo: window.trailingAnchor, constant: -12).isActive = true
+        }
+    }
+    
+    func dismissView() {
+        let animationDistance = self.window?.frame.height
+        UIView.animate(withDuration: 1.0, delay: 0, options: [.allowUserInteraction], animations: {
+            self.bannerView?.transform = CGAffineTransform(translationX: 0, y: animationDistance ?? CGFloat.greatestFiniteMagnitude)
+        }, completion: {_ in
+            self.removeBanner()
```

Comment:
Is [weak self] needed for to avoid holding a strong reference of self?

Reply:
good catch!

Reply:
good point, updated!